src/JsonSerialisation.ts

```diff
@@ -48,16 +41,214 @@ export class JsonSerialisation{
+			writer.WriteIntProperty(key, value);
+		}
+		writer.WriteObjectEnd();
+	}
+
+	public static WriteRuntimeObject(writer: SimpleJson.Writer, obj: InkObject): void{
+		// var container = obj as Container;
+		let container = asOrNull(obj, Container);
+		if (container) {
+			return this.WriteRuntimeContainer(writer, container);
+		}
+
+		// var divert = obj as Divert;
+		let divert = asOrNull(obj, Divert);
+		if (divert) {
+			let divTypeKey = '->';
+			if (divert.isExternal)
+				divTypeKey = 'x()';
+			else if (divert.pushesToStack) {
+				if (divert.stackPushType == PushPopType.Function)
+					divTypeKey = 'f()';
+				else if (divert.stackPushType == PushPopType.Tunnel)
+					divTypeKey = '->t->';
+			}
+
+			let targetStr;
+			if (divert.hasVariableTarget)
+				targetStr = divert.variableDivertName;
+			else
+				targetStr = divert.targetPathString;
+
+			writer.WriteObjectStart();
+			writer.WriteProperty(divTypeKey, targetStr);
+
+			if (divert.hasVariableTarget)
+				writer.WriteProperty('var', true);
+
+			if (divert.isConditional)
+				writer.WriteProperty('c', true);
+
+			if (divert.externalArgs > 0)
+				writer.WriteIntProperty('exArgs', divert.externalArgs);
+
+			writer.WriteObjectEnd();
+		}
+
+		// var choicePoint = obj as ChoicePoint;
+		let choicePoint = asOrNull(obj, ChoicePoint);
+		if (choicePoint) {
+			writer.WriteObjectStart();
+			writer.WriteProperty('*', choicePoint.pathStringOnChoice);
+			writer.WriteIntProperty('flg', choicePoint.flags);
+			writer.WriteObjectEnd();
+		}
+
+		// var intVal = obj as IntValue;
+		let intVal = asOrNull(obj, IntValue);
+		if (intVal)
+			writer.WriteInt(intVal.value);
+
+		// var floatVal = obj as FloatValue;
+		let floatVal = asOrNull(obj, FloatValue);
+		if (floatVal)
+			writer.WriteFloat(floatVal.value);
+
+		// var strVal = obj as StringValue;
+		let strVal = asOrNull(obj, StringValue);
+		if (strVal) {
+			if (strVal.isNewline)
+				writer.Write('\n', false);
+			else
+			writer.WriteStringStart();
+			writer.WriteStringInner('^');
+			writer.WriteStringInner(strVal.value);
+			writer.WriteStringEnd();
+		}
+
+		// var listVal = obj as ListValue;
+		let listVal = asOrNull(obj, ListValue);
+		if (listVal) {
+			this.WriteInkList(writer, listVal);
 		}
 
-		return jsonObj;
+		// var divTargetVal = obj as DivertTargetValue;
+		let divTargetVal = asOrNull(obj, DivertTargetValue);
+		if (divTargetVal) {
+			writer.WriteObjectStart();
+			if (divTargetVal.value === null) { return throwNullException('divTargetVal.value'); }
+			writer.WriteProperty('^->', divTargetVal.value.componentsString);
+			writer.WriteObjectEnd();
+
+			return;
+		}
+
+		// var varPtrVal = obj as VariablePointerValue;
+		let varPtrVal = asOrNull(obj, VariablePointerValue);
+		if (varPtrVal) {
+			writer.WriteObjectStart();
+			writer.WriteProperty('^var', varPtrVal.value);
+			writer.WriteIntProperty('ci', varPtrVal.contextIndex);
+			writer.WriteObjectEnd();
+			return;
+		}
+
+		// var glue = obj as Runtime.Glue;
```

Comment:
Can be removed too, and there are some more further down :)

Reply:
Haha, yeah, remnants of the pre-TS era.